nova_vm/src/ecmascript/builtins/indexed_collections/array_objects/array_prototype.rs

```diff
@@ -2088,8 +2088,68 @@ impl ArrayPrototype {
         Ok(accumulator)
     }
 
-    fn reverse(_agent: &mut Agent, _this_value: Value, _: ArgumentsList) -> JsResult<Value> {
-        todo!()
+    fn reverse(agent: &mut Agent, this_value: Value, _: ArgumentsList) -> JsResult<Value> {
+        // 1. Let O be ? ToObject(this value).
+        let o = to_object(agent, this_value)?;
```

Comment:
**suggestion:** Here it would be a good idea to add a fast path for "trivial" Arrays, see `is_trivial` check usage in this file. I haven't checked but I assume Rust has a reverse-in-place kind of function for mut slices, which we can safely use with trivial Arrays.

Reply:
Hey @aapoalas, thanks for your suggestion. I just added the fast path :)